src/client_side.cc

```diff
@@ -2275,7 +2283,7 @@ ConnStateData::requestTimeout(const CommTimeoutCbParams &io)
     if (!Comm::IsConnOpen(io.conn))
         return;
 
-    if (mayTunnelUnsupportedProto() && !receivedFirstByte_) {
+    if (!receivedFirstByte_ && preserveDataForTunnellingUnsupportedProto()) {
```

Comment:
I would expect to see an ableToTunnelUnsupportedProto() check here. preserveDataForTunnellingUnsupportedProto() decides whether to preserve data. We should have decided that earlier, _before_ the timeout. Here, after a timeout, we can only check whether the preconditions for tunneling are satisfied (probably _including_ the !receivedFirstByte_ precondition, but I am not sure about that):

```suggestion
    if (ableToTunnelUnsupportedProto()) {
```

Reply:
Well. You are right, however not easy to find all earlier stages where the decision should taken. Moreover not easy to make a function preserveDataForTunnellingUnsupportedProto() which should take the decision in various cases.

In the latest commit, I am setting directly the ConnStateData::ableToTunnelUnsupportedProto_  inside ConnStateData::switchToHttps (after this is called we are expecting TLS hello data). This member also set directly inside ConnStateData::parseTlsHandshake(), and it is reset after the first byte received (inside ConnStateData::receivedFirstByte).

I renamed the preserveDataForTunnellingUnsupportedProto()  to be a preserveHttpForTunnellingUnsupportedProto()  and it is now used only while we are expecting plain HTTP requests. It is used:
  * Http::One::Server::start() to detect timeouts before any HTTP request is received inside bumped connections or in intercepted plain HTTP connections
  * and in Http::One::Server::parseOneRequest to recognise the 1st HTTP request on bumped connections or in intercepted plain HTTP connections.

Please check my commit e757c76